prow/plugins/updateconfig/updateconfig.go

```diff
@@ -128,33 +136,33 @@ func handle(gc githubClient, kc kubeClient, log *logrus.Entry, pre github.PullRe
 		return err
 	}
 
-	var msg string
+	var updated []string
 	for _, change := range changes {
-		if change.Filename == configFile {
-			if err := handleConfig(gc, kc, org, repo, *pr.MergeSHA, configFile); err != nil {
-				return err
-			}
-
-			msg += fmt.Sprintf("I updated Prow config for you!")
-
-		} else if change.Filename == pluginFile {
-			if err := handlePlugin(gc, kc, org, repo, *pr.MergeSHA, pluginFile); err != nil {
-				return err
-			}
-
-			if msg != "" {
-				msg += "\n--------------------------\n"
-			}
-
-			msg += fmt.Sprintf("I updated Prow plugins config for you!")
+		cm, ok := configMaps[change.Filename]
+		if !ok {
+			continue
+		}
+		if err := update(gc, kc, org, repo, *pr.MergeSHA, change.Filename, cm); err != nil {
+			return err
 		}
+		updated = append(updated, cm)
 	}
 
-	if msg != "" {
-		if err := gc.CreateComment(org, repo, pr.Number, plugins.FormatResponseRaw(pr.Body, pr.HTMLURL, pr.User.Login, msg)); err != nil {
-			return fmt.Errorf("comment err: %v", err)
+	var msg string
+	switch n := len(updated); n {
+	case 0:
+		return nil
```

Comment:
seems it's impossible to get 0 updated?

Reply:
Not necessarily we could want to update `plugins.yaml` but the PR might have only changed `foo.go` in which case `len(updated) == 0`